doc/source/docs/howto_build_docs.rst

```diff
@@ -4,76 +4,100 @@
 Building the NumPy API and reference docs
 =========================================
 
-We currently use Sphinx_ for generating the API and reference
-documentation for NumPy.  You will need Sphinx >= 2.2.0.
-
 If you only want to get the documentation, note that pre-built
 versions can be found at
 
     https://numpy.org/doc/
 
 in several different formats.
 
-.. _Sphinx: http://www.sphinx-doc.org/
 
+Prerequisites
+-------------
 
-Instructions
-------------
+Before proceeding further it should be noted that the documentation is built with the ``make`` tool,
+which is not available on Windows. It is recommended for Windows users to set up their development
+environment on :ref:`Gitpod <development-gitpod>` or `Windows Subsystem
+for Linux (WSL) <https://docs.microsoft.com/en-us/windows/wsl/install-win10>`_.
+Gitpod is recommended if you have good internet connectivity and want a temporary setup.
+WSL is a good option for a persistent local setup.
+
+Building the NumPy documentation and API reference requires the following:
+
+1) NumPy
+
+   Since large parts of the main documentation are obtained from NumPy via
+   ``import numpy`` and examining the docstrings, you will need to first
+   :ref:`build <building-from-source>`, and install it so that the correct version is imported.
+   NumPy has to be re-built and re-installed every time you fetch the latest version of the
+   repository, before generating the documentation. This ensures that the NumPy version and
+   the git repository version are in sync.
 
-If you obtained NumPy via git, get also the git submodules that contain
-additional parts required for building the documentation::
+   Note that you can eg. install NumPy to a temporary location and set
+   the PYTHONPATH environment variable appropriately.
+   Alternatively, if using Python virtual environments (via e.g. ``conda``,
+   ``virtualenv`` or the ``venv`` module), installing NumPy into a
+   new virtual environment is recommended.
 
-    git submodule update --init
+2) Dependencies
 
-In addition, building the documentation requires the Sphinx extension
-`plot_directive`, which is shipped with Matplotlib_. This Sphinx extension can
-be installed by installing Matplotlib. You will also need Python>=3.6.
+   All of the necessary dependencies for building the NumPy docs can be installed
+   with::
 
-Since large parts of the main documentation are obtained from numpy via
-``import numpy`` and examining the docstrings, you will need to first build
-NumPy, and install it so that the correct version is imported.
+       pip install -r doc_requirements.txt
 
-After NumPy is installed, install SciPy since some of the plots in the random
-module require `scipy.special` to display properly.
+   We currently use Sphinx_ (>= 2.2.0) for generating the API and reference
+   documentation for NumPy. In addition, building the documentation requires
```

Comment:
Actually, in the `doc_requirements.txt` file, we are asking for `sphinx==4.0.1`, but (surprisingly) in `environment.yml` we are using `sphinx=3.5.4`. I'm assuming anything above 3.5.4 works - @rossbar can you help us here?

Reply:
I didn't update the `environment.yml` when I updated `doc_requirements.txt`, and we have no check that the requirements are in sync. It would be nice to have an automatic way to update the docs, maybe we could `include` the relevant lines of the `doc_requirements.txt` or alternatively suggest users do `pip install -r doc_requirements.txt` without mentioning exact versions?

Reply:
I think not specifying the exact versions will be better for the maintenance of the doc. Since users don't have to separately install `Sphinx`, I don't think it will cause any problems.

Reply:
> I'm assuming anything above 3.5.4 works

AFAIK there's no reason to pin to 3.5.4 - I've been building with ~=4 for a while now without problems

> and we have no check that the requirements are in sync.

I'm not sure what the best practices are for keeping `environments.yml` and `requirements.txt` in sync. I think there's a way in `environments.yml` files to specify to use `pip` for certain dependencies... maybe that supports requirements files? I'm not really sure...